src/Mod/TechDraw/Gui/TaskSurfaceFinishSymbols.cpp

```diff
@@ -336,11 +361,14 @@ void TaskSurfaceFinishSymbols::onIconChanged()
     if (ui->pbIcon06 == pressedButton) activeIcon = removeRequiredAll;
 
     QIcon symbolIcon = pressedButton->icon();
-    QGraphicsPixmapItem* pixmapItem = new(QGraphicsPixmapItem);
-    pixmapItem->setPixmap(symbolIcon.pixmap(50, 64));
-    pixmapItem->setPos(-50, -130);
-    pixmapItem->setZValue(-1);
-    symbolScene->addItem(pixmapItem);
+    if(currentIcon) {
+        symbolScene->removeItem(currentIcon);
+    }
+    currentIcon = new(QGraphicsPixmapItem);
```

Comment:
Hmmm, would this cause a memory leak?

Reply:
If it does not have parent set properly it can.